src/agent/src/policy.rs

```diff
@@ -120,13 +154,56 @@ impl AgentPolicy {
         let query = format!("data.agent_policy.{ep}");
         self.engine.set_input_json(ep_input)?;
 
-        let mut allow = match self.engine.eval_bool_query(query, false) {
-            Ok(a) => a,
-            Err(e) => {
-                if !self.allow_failures {
-                    return Err(e);
+        let results = self.engine.eval_query(query, false)?;
+
+        let prints = match self.engine.take_prints() {
+            Ok(p) => p.join(" "),
+            Err(e) => format!("Failed to get policy log: {e}"),
+        };
+
+        if results.result.len() != 1 {
+            // Results are empty for UpdateInterfaceRequest when AllowRequestsFailingPolicy is set to true
```

Comment:
This other comment seems more useful:

Results are empty when AllowRequestsFailingPolicy is used to allow a Request that hasn't been defined in the policy.

Reply:
Updated